.github/workflows/AssignReviewers.yml

```diff
@@ -0,0 +1,31 @@
+## @file
+# Assign reviewers from a REVIEWERS file using CODEOWNERS syntax
+#
+# Copyright (c) 2022, Intel Corporation. All rights reserved.<BR>
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+# Copied from https://github.com/mdkinney/github-action-assign-reviewers
+# commit 77731e1
+
+name: Assign reviewers from a REVIEWERS file using CODEOWNERS syntax
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened, ready_for_review]
+    branches:
+      - main
+
+jobs:
+  assign_reviewers:
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout Pull Request Target
+        uses: actions/checkout@v2

```

Comment:
It would be nice to try a sparse checkout if possible.

Reply:
A full checkout doesn't take very long though, and sparse-checkout is marked experimental. Do you think it's worth trying?

Reply:
I've used the same edk2-platforms repo for a long time, so I don't have a good reference for how long it's taking. If it is not actually that bad (>10s or so), then it's not going to help much.

Reply:
Hmm, do you mean a sparse/shallow/partial _clone_ vs. a full _clone_?
A `git checkout` of edk2 with an existing clone going from the `vUDK2017` tag all the way up to the latest `master` takes around 2 seconds.

Reply:
Yes. For example, a full checkout of edk2 before I switched to sparse would take around 20s - https://github.com/tianocore/edk2/actions/runs/10251696000/job/28360319318.

After, it is now around 1s - https://github.com/tianocore/edk2/actions/runs/10261522158/job/28389416364.

I did have `fetch-depth` set to `0` before which definitely had an impact as well. But, a minimal fetch depth combined with minimal file checkout helps. If the workflow itself is checking out around 2s then you don't need to worry about it.

Do you have a fork or somewhere you've tested it?

Reply:
I still need to figure out how to test it.
It was the fetch + checkout that took 20s! For me, `git -c protocol.version=2 fetch --prune --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*` takes about 30 seconds, while the `git checkout --progress --force -B master refs/remotes/origin/master` takes a fraction of a second.

Reply:
@makubacki I've been trying to test it: my latest attempt is https://github.com/bcran/edk2-platforms/pull/2, which failed because it can't assign me to the PR (presumably because it's in my account).

Maybe we could merge this and fix any problems as they are found?

Reply:
Yes, the author of the PR cannot be added as a reviewer. I used some bot accounts for testing in my fork.

I don't have a problem with merging it to allow testing to complete until an official transition to PRs in the repo. I assume most of the underlying action logic has already been tested that is being wrapped here.

@mdkinney for his opinion.

Reply:
With https://github.com/bcran/edk2-platforms/actions/runs/10647138622/job/29514752629?pr=3 it's clearly parsing the REVIEWERS file, seeing that my bot account (@bexcran-ghbot01) should be assigned to the PR and is just failing at the last step.

I think we should be good to go with these changes.

Reply:
@mdkinney, @makubacki Could you approve this PR please?